src/Microsoft.Health.Fhir.Core/Messages/Search/CompartmentResourceRequest.cs

```diff
@@ -0,0 +1,30 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using MediatR;
+
+namespace Microsoft.Health.Fhir.Core.Messages.Search
+{
+    public class CompartmentResourceRequest : IRequest<CompartmentResourceResponse>
+    {
+        public CompartmentResourceRequest(string compartmentType, string compartmentId, string resourceType, IReadOnlyList<Tuple<string, string>> queries)
```

Comment:
can these be null?

Reply:
The route constraints validate for proper compartment and resource types. The core layer is resilient for these values to be null.